libs/blocks/mas-autoblock/mas-autoblock.js

```diff
@@ -22,32 +50,162 @@ export function getTagName(el) {
   return el.textContent.trim().match(/^[^:\s]+/)?.[0] || 'merch-card';
 }
 
-export async function createCard(el, fragment) {
-  // add <mas-commerce-service>
+async function loadControl(tagName) {
+  /** Load service first */
   const servicePromise = initService();
-  const timeoutPromise = new Promise((resolve) => {
-    setTimeout(() => resolve(false), MAS_AUTOBLOCK_TIMEOUT);
-  });
-  let success = await Promise.race([servicePromise, timeoutPromise]);
+  const success = await Promise.race([servicePromise, getTimeoutPromise()]);
   if (!success) {
     throw new Error('Failed to initialize mas commerce service');
   }
   const service = await servicePromise;
   log = service.Log.module('merch');
 
-  const aemFragment = createTag('aem-fragment', { fragment });
-  const merchCard = createTag(getTagName(el), { consonant: '' }, aemFragment);
-  el.replaceWith(merchCard);
-  const merchCardPromise = merchCard.checkReady();
-  success = await Promise.race([merchCardPromise, timeoutPromise]);
+  /** Load the control and any dependencies */
+  const { base } = getConfig();
+  switch (tagName) {
+    case 'merch-card-collection':
+      await import('../../deps/mas/merch-card-collection.js');
+      await import('../../deps/mas/merch-sidenav.js');
+      await import(`${base}/features/spectrum-web-components/dist/theme.js`);
+      await import(`${base}/features/spectrum-web-components/dist/button.js`);
+      await import(`${base}/features/spectrum-web-components/dist/action-button.js`);
+      await import(`${base}/features/spectrum-web-components/dist/action-menu.js`);
+      await import(`${base}/features/spectrum-web-components/dist/search.js`);
+      await import(`${base}/features/spectrum-web-components/dist/sidenav.js`);
+      await import(`${base}/features/spectrum-web-components/dist/menu.js`);
+      await import(`${base}/features/spectrum-web-components/dist/checkbox.js`);
+      await import(`${base}/features/spectrum-web-components/dist/dialog.js`);
+      await import(`${base}/features/spectrum-web-components/dist/link.js`);
+      await import(`${base}/features/spectrum-web-components/dist/overlay.js`);
+      await import(`${base}/features/spectrum-web-components/dist/tray.js`);
+      await import(`${base}/features/spectrum-web-components/dist/shared.js`);
+      await import(`${base}/features/spectrum-web-components/dist/base.js`);
+      break;
+    default:
+      break;
+  }
+}
+
+/**
+ * @param {string} fragment
+ * @param {string} tagName
+ */
+function getTagAttributesAndContent(fragment, tagName) {
+  let attributes;
+  const html = createTag('aem-fragment', { fragment });
+  switch (tagName) {
```

Comment:
nit:
this also could be just
```
if (tagName === 'merch-card') {
    attributes = { consonant: '' };
}
```
3 lines instead of 7 and easier to read